scc/tests/args.rs

```diff
@@ -25,226 +36,166 @@ const TEST_DIRECTORY: TestArg = TestArg {
-    }
-    #[test]
-    fn test_fdfd() {
-        test_order(&[TEST_FILE, TEST_DIRECTORY, TEST_DIRECTORY, TEST_FILE]);
-    }
-    #[test]
-    fn test_fddfd() {
-        test_order(&[TEST_FILE, TEST_DIRECTORY, TEST_DIRECTORY, TEST_FILE, TEST_DIRECTORY]);
-    }
-    #[test]
-    fn test_df() {
-        test_order(&[TEST_DIRECTORY, TEST_FILE]);
-    }
-    #[test]
-    fn test_dfd() {
-        test_order(&[TEST_DIRECTORY, TEST_FILE, TEST_DIRECTORY]);
-    }
-    #[test]
-    fn test_ddf() {
-        test_order(&[TEST_DIRECTORY, TEST_DIRECTORY, TEST_FILE]);
-    }
-    #[test]
-    fn test_dff() {
-        test_order(&[TEST_DIRECTORY, TEST_FILE, TEST_FILE]);
-    }
-    #[test]
-    fn test_dffd() {
-        test_order(&[TEST_DIRECTORY, TEST_FILE, TEST_FILE, TEST_DIRECTORY]);
-    }
-    #[test]
-    fn test_dfdf() {
-        test_order(&[TEST_DIRECTORY, TEST_FILE, TEST_FILE, TEST_DIRECTORY]);
+    fn get_help() {
+        println!(
+            "{}",
+            Opts::get_parser()
+                .run_inner(bpaf::Args::from(&["--help"]))
+                .unwrap_err()
+                .unwrap_stdout()
+        );
     }
+
     #[test]
-    fn test_dffdf() {
-        test_order(&[TEST_DIRECTORY, TEST_FILE, TEST_FILE, TEST_DIRECTORY, TEST_FILE]);
+    fn check_options() {
+        Opts::get_parser().check_invariants(false)
     }
-}
 
-mod cybercmd {
-    use super::*;
+    #[apply(file_directory_orders)]
+    fn standard(
+        f_args: &[TestArg],
+        cache_file: Option<&TestArg>,
+        cache_dir: Option<&TestArg>,
+        script_paths_file: Option<&TestArg>,
+        warnings: Option<&str>,
+        optimize: Option<bool>,
+        threads: Option<u8>,
+        no_testonly: Option<bool>,
+        no_breakpoint: Option<bool>,
+        profile: Option<bool>,
+    ) {
+        std::panic::set_hook(Box::new(|a| println!("{:#?}", a)));
+        let mut expected_args = vec![
+            "-compile",
+            "C:\\Program Files (x86)\\Steam\\steamapps\\common\\Cyberpunk 2077\\r6\\scripts\\",
+        ];
+        let mut arg_str =
+            String::from(r#"-compile "C:\Program Files (x86)\Steam\steamapps\common\Cyberpunk 2077\r6\scripts\""#);
+        if let Some(value) = cache_file {
+            expected_args.push(value.parsed);
+            arg_str.push_str(value.raw);
+        }
+        if let Some(value) = cache_dir {
+            expected_args.push("-customCacheDir");
+            expected_args.push(value.parsed);
+            arg_str.push_str(value.raw);
+        }
+        if let Some(value) = script_paths_file {
+            expected_args.push("-compilePathsFile");
+            expected_args.push(value.parsed);
+            arg_str.push_str(value.raw);
+        }
+        let warning_str;
+        if let Some(value) = warnings {
+            warning_str = ["-W", value].join("");
+            expected_args.push(&warning_str);
+            arg_str.push_str(" -W");
+            arg_str.push_str(value);
+        }
```

Comment:
```suggestion
        if let Some(value) = warnings {
            expected_args.push(format!("-W{value}"));
            arg_str.push_str(" -W");
            arg_str.push_str(value);
        }
```
I'd get rid of the intermediate `*_str` locals and just push owned strings to `expected_args` and when you're inserting a &str just call `to_owned` on it, it'll simplify this function a lot

Also, what's the point of `arg_str` and how is it different from just `expected_args.join(" ")`?

Reply:
ok cool - I think I can convert it all over to use one.